packages/auth-sveltekit/src/server.ts

```diff
@@ -0,0 +1,489 @@
+import {
+  redirect,
+  type Cookies,
+  type RequestEvent,
+  type Handle,
+} from "@sveltejs/kit";
+import type { Client } from "edgedb";
+import {
+  Auth,
+  builtinOAuthProviderNames,
+  type BuiltinOAuthProviderNames,
+  type TokenData,
+  type emailPasswordProviderName,
+} from "@edgedb/auth-core";
+import {
+  ClientAuth,
+  getConfig,
+  type AuthConfig,
+  type AuthOptions,
+} from "./client.js";
+
+export type { TokenData, AuthOptions, Client };
+
+export type BuiltinProviderNames =
+  | BuiltinOAuthProviderNames
+  | typeof emailPasswordProviderName;
+
+type ParamsOrError<Result extends object, ErrorDetails extends object = {}> =
+  | ({ error: null } & { [Key in keyof ErrorDetails]?: undefined } & Result)
+  | ({ error: Error } & ErrorDetails & { [Key in keyof Result]?: undefined });
+
+export interface AuthRouteHandlers {
+  onOAuthCallback(
+    params: ParamsOrError<{
+      tokenData: TokenData;
+      provider: BuiltinOAuthProviderNames;
+      isSignUp: boolean;
+    }>
+  ): Promise<Response>;
+  onBuiltinUICallback(
+    params: ParamsOrError<
+      (
+        | {
+            tokenData: TokenData;
+            provider: BuiltinProviderNames;
+          }
+        | { tokenData: null; provider: null }
+      ) & { isSignUp: boolean }
+    >
+  ): Promise<Response>;
+  onEmailVerify(
+    params: ParamsOrError<
+      { tokenData: TokenData },
+      { verificationToken?: string }
+    >
+  ): Promise<Response>;
+  onSignout(): Promise<Response>;
+}
+
+export default function serverAuth(client: Client, options: AuthOptions) {
+  const core = Auth.create(client);
+  const config = getConfig(options);
+
+  return {
+    createServerRequestAuth: ({ event }: { event: RequestEvent }) =>
+      new ServerRequestAuth(client, core, event, options),
+    createAuthRouteHook:
+      (handlers: AuthRouteHandlers): Handle =>
+      ({ event, resolve }) => {
+        const pathname = new URL(event.request.url).pathname;
+
+        if (pathname.startsWith(`/${config.authRoutesPath}`)) {
```

Comment:
```suggestion
        if (pathname.startsWith(`/${config.authRoutesPath}/`)) {
```
Just in case the user has some other route that starts with the word 'auth' (or whatever they've set the auth route as).